libraries/AP_Scripting/AP_Scripting.cpp

```diff
@@ -281,6 +281,31 @@ void AP_Scripting::handle_mission_command(const AP_Mission::Mission_Command& cmd
     mission_data->push(cmd);
 }
 
+bool AP_Scripting::arming_checks(size_t buflen, char *buffer) const
+{
+    if (!enabled() || ((_debug_options.get() & uint8_t(lua_scripts::DebugLevel::DISABLE_PRE_ARM)) != 0)) {
+        return true;
+    }
+
+    if (_thread_failed) {
+        hal.util->snprintf(buffer, buflen, "Scripting: %s", "failed to start");
+        return false;
+    }
+
+    if (_init_failed) {
+        hal.util->snprintf(buffer, buflen, "Scripting: %s", "out of memory");
+        return false;
+    }
+
+    char *error_buf = lua_scripts::get_error_last_message();
```

Comment:
Note that were now accessing this from a different thread than the one that populates it. I think its harmless, the worst case would be a nonsense message of `buflen`. We could add a semaphore?